### PR TITLE
DOCS-6008 connections

### PR DESCRIPTION
**Pull request by sathya0**

### CODE REVIEW

File: modules/ROOT/pages/connections.adoc [hunk 1]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
...
+Connected modules (connectors) are the most common type of module. They provide a connection between a Mule flow and an external resource.
+Configure these connections with the `ConnectionProvider` in a connectivity framework to create, disconnect, and describe the connection for Mule runtime engine to handle.
+For more information on how to use connections, refer to <<use-connections, this section>>.
 
 == Declaring a Connection Provider
 
 To declare a new `ConnectionProvider` you should create a new class that implements the `ConnectionProvider` interface. The `ConnectionProvider` interface is a typed interface that receives one type argument that indicates the class of the instances that the `ConnectionProvider` handles.
 
-This example declares a `FooConnectionProvider` class that implements the `ConnectionProvider` interface:
+This example declares a `myConnectionProvider` class that implements the `ConnectionProvider` interface:
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: by naming convention, class name should start with capital letter, so I'd replace _myConnectionProvider_ with _MyConnectionProvider_

File: modules/ROOT/pages/connections.adoc [hunk 2]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
...
 
 [source,java,linenums]
 ----
-public class FooConnectionProvider implements ConnectionProvider<FooConnection> {
+public class myConnectionProvider implements ConnectionProvider<myConnection> {
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: same here:

public class MyConnectionProvider implements ConnectionProvider<MyConnection> {

File: modules/ROOT/pages/connections.adoc [hunk 3]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
...
 
   @Parameter
   private String password;
 
-  public FooConnection connect() throws ConnectionException {
-    return new FooConnection(password);
+  public myConnection connect() throws ConnectionException {
+    return new myConnection(password);
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: replace _myConnection_ with _MyConnection_

File: modules/ROOT/pages/connections.adoc [hunk 4]
```diff
@@ -91,10 +90,10 @@ Example of a `ConnectionProvider` declared at `@Extension` level:
 
 [source,java,linenums]
 ----
-@Extension(name="foo")
+@Extension(name="xyz")
 @Operations(Operations.class)
-@ConnectionProviders(FooConnectionProvider.class)
-public class FooModule {
+@ConnectionProviders(myConnectionProvider.class)
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: replace by MyConnectionProvider

File: modules/ROOT/pages/connections.adoc [hunk 5]
```diff
@@ -91,10 +90,10 @@ Example of a `ConnectionProvider` declared at `@Extension` level:
...
+public class myModule {
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: replace with MyModule

File: modules/ROOT/pages/connections.adoc [hunk 6]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
...
   private String password;
 
-  public FooConnection connect() throws ConnectionException {
-    return new FooConnection(password);
+  public myConnection connect() throws ConnectionException {
+    return new myConnection(password);
   }
 
-  public void disconnect(FooConnection connection) {
+  public void disconnect(myConnection connection) {
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: replace with MyConnection

File: modules/ROOT/pages/connections.adoc [hunk 7]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
...
     connection.invalidate();
   }
 
-  public ConnectionValidationResult validate(FooConnection connection) {
+  public ConnectionValidationResult validate(myConnection connection) {
```
Comment (dabreu) [inline modules/ROOT/pages/connections.adoc]: replace with MyConnection

**Review by dabreu — DISMISSED** [review]

File: modules/ROOT/pages/connections.adoc [hunk 8]
```diff
@@ -4,33 +4,32 @@ include::_attributes.adoc[]
 endif::[]
 :keywords: mule, sdk, connection, connectivity, management, pooling, cached, provider, connection-provider
 
-Connected modules (connectors) are probably the most common type of module.
-
-The connectivity framework starts with the concept of a `ConnectionProvider`.
-This is an interface responsible for creating, disconnecting, and describing the connection so that the runtime can handle it properly, while remaining abstracted from the concerns of managing those connections.
+Connected modules (connectors) are the most common type of module. They provide a connection between a Mule flow and an external resource.
+Configure these connections with the `ConnectionProvider` in a connectivity framework to create, disconnect, and describe the connection for Mule runtime engine to handle.
+For more information on how to use connections, refer to <<use-connections, this section>>.
```
Comment (hannanelson) [inline modules/ROOT/pages/connections.adoc]:
```suggestion
For more information about how to use connections, refer to <<use-connections>>.
```

Comment (hannanelson) [reply modules/ROOT/pages/connections.adoc]: (Name the section, rather than using "this section".)

Comment (hannanelson) [reply modules/ROOT/pages/connections.adoc]: Kim would say use "about" not "on" :)

**Review by hannanelson — DISMISSED** [review]

File: modules/ROOT/pages/connections.adoc [hunk 9]
```diff
@@ -143,9 +142,10 @@ connection-handling behavior. An operation that requires a connection calls the
...
 
+[[use-connections]]
 == Using Connections
 
 Using connections in Operations and Sources is quite different, see:
 
 * <<operations#_using_connections, Using Connections in Operations>>
-* <<sources-config-connection#, Using Connections in Sources>>
\ No newline at end of file
+* <<sources-config-connection#, Using Connections in Sources>>
```
Comment (hannanelson) [inline modules/ROOT/pages/connections.adoc:151]: What is the # doing in this case? (I've never seen that construction.)

**Review by hannanelson — APPROVED** [review]

**Pull request merged**